planning_through_contact/simulation/controllers/desired_position_source_base.py

```diff
@@ -0,0 +1,15 @@
+from abc import ABC, abstractmethod
+
+from pydrake.all import DiagramBuilder, Meshcat, OutputPort
+
+
+class DesiredPositionSourceBase(ABC):
+    """The desired position source base class."""
+
+    @abstractmethod
+    def setup(self, builder: DiagramBuilder, **kwargs) -> OutputPort:
```

Comment:
Drake convention for adding a system to a diagram with the DiagramBuilder is `AddToBuilder`. If this is what this function is doing, I suggest we change the name.

Reply:
This is slightly unideally written because the Position Source and Position Controllers are not actually themselves systems but once we fix all the other problems I will come back and refactor these to make them systems I think that will be cleaner. But in the meantime I've changed the name of setup to AddToBuilder.